src/slots/AnyInSlot.js

```diff
@@ -0,0 +1,45 @@
+const Joi = require('joi')
+
+const JOIous = require('../mixins/instanceMixins/JOIous')
+const InSlot = require('./InSlot')
+const SlotConnectionError = require('./SlotConnectionError')
+
+class AnyInSlot extends InSlot {
+
+  static get DATA_TYPE() {
+    return super.ANY_DATA_TYPE
+  }
+
+  static get SCHEMA() {
+    return super.SCHEMA.concat(Joi.object({
+      dataType: Joi.string().allow(this.DATA_TYPE).required(),
+    }))
+  }
+
+  connectTo(otherSlot, dataStreamData) {
+    /* datastream default values (e.g. averagingWindowSize) depend on the dataType
+     * of the slot; when connectTo is called on the "any" dataType slot, we simply
+     * check if the other slot is not of dataType "any" and if so, we simply call connnectTo
+     * on the other slot, which then will set the proper default values;
+     *
+     * the default case prevents an infinite loop when connecting two "any" slots
+     */
+    if (otherSlot.type === InSlot.TYPE) {
+      throw new SlotConnectionError(this, otherSlot, 'must have complimentary types')
+    }
+
+    if (otherSlot.dataType !== AnyInSlot.DATA_TYPE) {
+      return otherSlot.connectTo(this, dataStreamData)
+    }
+
+    const extendedDataStreamData = {
+      averagingWindowSize: 1,
+      ...dataStreamData,
+    }
+    return super.connectTo(otherSlot, extendedDataStreamData)
+
+  }
```

Comment:
Not happy about the duplication, but there is no easy way to fix this in the current class hierarchy: Slots neeed to inherit behaviour once from their direction "InSlot" or "OutSlot" and we would also need something like "AnySlot", "BooleanSlot". Too bad JS does not support multi-inheritance. Functional programming and composition could work here, maybe mixins...